temporalio/client.py

```diff
@@ -1858,8 +1859,8 @@ async def execute_update(
             update,
             arg,
             args=args,
+            wait_for_stage=WorkflowUpdateWaitStage.COMPLETED,
```

Comment:
Just want to double-check we're sure `WorkflowUpdateWaitStage` is the right name for the  enum here. The underlying API entity is called "Lifecycle Stage". Calling it "Wait Stage" bakes the concept of waiting into it -- is there a chance that that will be annoying in the future because we find ourselves wanting to refer to a lifecycle stage without any notion of "waiting" being involved?

Reply:
:+1: This is the value of this PR I think. I have no strong opinion here and I now that I think about it, I like `WorkflowUpdateLifecycleStage` because, when we someday get a `describe()` call on `WorkflowUpdateHandle`, we'll want this stage to be there too.

Anyone else object to me changing this enum name to `WorkflowUpdateLifecycleStage`?

Reply:
Yeah we should consider the future `describe()` call, which we can implement today, so calling the enum `WorkflowUpdateLifecycleStage` makes a lot more sense.

Reply:
Updated to `WorkflowUpdateLifecycleStage`

Reply:
+1 `Lifecycle` isn't needed -- `WorkflowUpdateStage` is sufficient.

Reply:
Enum name updated